lib/services/local/user_integration_task.go

```diff
@@ -0,0 +1,85 @@
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package local
+
+import (
+	"context"
+
+	"github.com/gravitational/trace"
+
+	userintegrationtasksv1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/userintegrationtasks/v1"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/backend"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/teleport/lib/services/local/generic"
+)
+
+type UserIntegrationTasksService struct {
+	service *generic.ServiceWrapper[*userintegrationtasksv1.UserIntegrationTask]
+}
+
+const userIntegrationTasksKey = "user_integration_tasks"
+
+// NewUserIntegrationTasksService creates a new UserIntegrationTasksService.
+func NewUserIntegrationTasksService(backend backend.Backend) (*UserIntegrationTasksService, error) {
+	service, err := generic.NewServiceWrapper(backend,
+		types.KindUserIntegrationTask,
+		userIntegrationTasksKey,
+		services.MarshalUserIntegrationTask,
+		services.UnmarshalUserIntegrationTask)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return &UserIntegrationTasksService{service: service}, nil
+}
+
+func (s *UserIntegrationTasksService) ListUserIntegrationTasks(ctx context.Context, pagesize int64, lastKey string) ([]*userintegrationtasksv1.UserIntegrationTask, string, error) {
+	r, nextToken, err := s.service.ListResources(ctx, int(pagesize), lastKey)
+	return r, nextToken, trace.Wrap(err)
+}
+
+func (s *UserIntegrationTasksService) GetUserIntegrationTask(ctx context.Context, name string) (*userintegrationtasksv1.UserIntegrationTask, error) {
+	r, err := s.service.GetResource(ctx, name)
+	return r, trace.Wrap(err)
+}
+
+func (s *UserIntegrationTasksService) CreateUserIntegrationTask(ctx context.Context, userIntegrationTask *userintegrationtasksv1.UserIntegrationTask) (*userintegrationtasksv1.UserIntegrationTask, error) {
+	r, err := s.service.CreateResource(ctx, userIntegrationTask)
+	return r, trace.Wrap(err)
+}
+
+func (s *UserIntegrationTasksService) UpdateUserIntegrationTask(ctx context.Context, userIntegrationTask *userintegrationtasksv1.UserIntegrationTask) (*userintegrationtasksv1.UserIntegrationTask, error) {
+	r, err := s.service.ConditionalUpdateResource(ctx, userIntegrationTask)
+	return r, trace.Wrap(err)
+}
+
+func (s *UserIntegrationTasksService) UpsertUserIntegrationTask(ctx context.Context, userIntegrationTask *userintegrationtasksv1.UserIntegrationTask) (*userintegrationtasksv1.UserIntegrationTask, error) {
+	r, err := s.service.UpsertResource(ctx, userIntegrationTask)
```

Comment:
Don't we usually have validation checks at the `lib/services/local` layer rather than the auth service layer?

Reply:
I think we use both places
I did the boilerplate code by following a resource that was validating at the auth service layer.

Let me know if you think this should be moved.
I don't think there's much of a difference given that the method ends up being called right away.

Reply:
If you leave it like this you're likely not going to get validation when putting values in or pulling values out from the cache (if this ends up being cached in the future), which is atypical, at the very least - it might be correct when pulling values out, but we should probably check them before storing them locally.

Reply:
Moved to the backend layer at `lib/services/local/user_task.go`